### PR TITLE
Serialize credential with fields names using only 1 bytes

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This saves space when serializing credentials.

We cannot use [`#[serde(remote)]`](https://serde.rs/remote-derive.html) because that would require supporting `#[serde(with)]` in `serde_indexed`, already used for `CredentialData`

This replaces #58 it should be 1 bytes larger per serialized fields, but is better than serde-indexed as it has les risks to break accidentally when adding/removing a field.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -76,9 +77,11 @@ ctap-types = { git = "https://github.com/trussed-dev/ctap-types.git", rev = "72e
 ctaphid-dispatch = { git = "https://github.com/trussed-dev/ctaphid-dispatch.git", rev = "57cb3317878a8593847595319aa03ef17c29ec5b" }
 apdu-dispatch = { git = "https://github.com/trussed-dev/apdu-dispatch.git", rev = "915fc237103fcecc29d0f0b73391f19abf6576de" }
 littlefs2 = { git = "https://github.com/trussed-dev/littlefs2.git", rev = "2b45a7559ff44260c6dd693e4cb61f54ae5efc53" }
+serde-indexed = { git = "https://github.com/sosthene-nitrokey/serde-indexed.git", rev = "5005d23cb4ee8622e62188ea0f9466146f851f0d" }
```
Comment (robin-nitrokey) [inline Cargo.toml:80]: Looks like this PR does not use `serde-indexed` (anymore).  Can we remove the patch?

File: Cargo.toml [hunk 2]
```diff
@@ -76,9 +77,11 @@ ctap-types = { git = "https://github.com/trussed-dev/ctap-types.git", rev = "72e
...
 trussed = { git = "https://github.com/trussed-dev/trussed.git", rev = "b548d379dcbd67d29453d94847b7bc33ae92e673" }
 trussed-chunked = { git = "https://github.com/trussed-dev/trussed-staging.git", tag = "chunked-v0.1.0" }
 trussed-hkdf = { git = "https://github.com/trussed-dev/trussed-staging.git", tag = "hkdf-v0.2.0" }
 trussed-staging = { git = "https://github.com/trussed-dev/trussed-staging.git", tag = "v0.3.0" }
 trussed-usbip = { git = "https://github.com/Nitrokey/pc-usbip-runner.git", tag = "v0.0.1-nitrokey.1" }
 usbd-ctaphid = { git = "https://github.com/trussed-dev/usbd-ctaphid.git", rev = "dcff9009c3cd1ef9e5b09f8f307aca998fc9a8c8" }
+cbor-smol = { git = "https://github.com/sosthene-nitrokey/cbor-smol.git", rev = "327f723d53263d4ac1da368660de4fe4aa8ebef8" }
```
Comment (robin-nitrokey) [inline Cargo.toml:87]: What change exactly is needed?  The referenced commit does not look related to this PR.

File: src/credential.rs [hunk 3]
```diff
@@ -205,15 +205,105 @@ impl Credential {
     }
 }
 
+/// Copy of [`ctap_types::webauthn::PublicKeyCredentialUserEntity`] but with `serde_indexed` serialization
```
Comment (robin-nitrokey) [inline src/credential.rs:208]:
```suggestion
/// Copy of [`ctap_types::webauthn::PublicKeyCredentialUserEntity`] but with shorter field names for serialization
```

File: src/credential.rs [hunk 4]
```diff
@@ -205,15 +205,105 @@ impl Credential {
...
+        Self {
+            id,
+            icon,
+            name,
+            display_name,
+        }
+    }
+}
+
+/// Copy of [`ctap_types::webauthn::PublicKeyCredentialUserEntity`] but with `serde_indexed` serialization
```
Comment (robin-nitrokey) [inline src/credential.rs:281]:
```suggestion
/// Copy of [`ctap_types::webauthn::PublicKeyCredentialUserEntity`] but with shorter field names for serialization
```

File: src/credential.rs [hunk 5]
```diff
@@ -205,15 +205,105 @@ impl Credential {
...
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct LocalPublicKeyCredentialUserEntity {
```
Comment (robin-nitrokey) [inline src/credential.rs:283]: nit: move struct definition above the impls

File: src/credential.rs [hunk 6]
```diff
@@ -724,6 +810,308 @@ mod test {
         });
     }
 
+    #[test]
+    fn indexed_derive_rp_name_none() {
```
Comment (robin-nitrokey) [inline src/credential.rs:814]: nit: s/indexed_derive/local/, also for the other tests

File: Cargo.toml (hunk 2 above)
Comment (sosthene-nitrokey) [reply Cargo.toml:87]: This was needed to support de-serializing integers as identifiers. Removed

File: CHANGELOG.md [hunk 7]
```diff
@@ -37,6 +38,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [#61]: https://github.com/Nitrokey/fido-authenticator/pull/61
 [#62]: https://github.com/Nitrokey/fido-authenticator/pull/62
 [#63]: https://github.com/Nitrokey/fido-authenticator/pull/63
+[#59]: ttps://github.com/Nitrokey/fido-authenticator/issues/59
```
Comment (robin-nitrokey) [inline CHANGELOG.md:41]: s/ttps/https/

**Review by robin-nitrokey — APPROVED** [review]

**Review by daringer — APPROVED** [review]
